edb/edgeql/compiler/__init__.py

```diff
@@ -505,6 +505,7 @@ def compile_func_to_ir(
     )
 
     assert isinstance(ir, irast.Statement)
+    schema = ir.schema
```

Comment:
Ha, I'm wondering why we haven't detected this before

Reply:
Because functions cannot produce schema objects other than themselves and collection types, which were previously handled differently.